File: frontend/nutri-facts-front/src/types/product.ts
export interface NutritionFacts {
  calories: number;
  carbohydrates: number;
  fat: number;
  proteins: number;
}

export interface Product {
  name: string;
  id: string;
  nutritionFacts: NutritionFacts;
}

export interface ProductEntry {
  id: string;
  product: Product;
  quantity: number;
  nutritionFacts: NutritionFacts;
}
export const format = new Intl.NumberFormat("pt-BR", {
  minimumFractionDigits: 0,
  maximumFractionDigits: 1,
});

File: backend/DTO/Product/Product.cs
//make this class serializable
public class Product : IProduct
{
    //Todo: make this field obligatory
    public string Id { get; set; } = null!;
    public string Name { get; set; } = null!;
    public string Brand { get; set; } = null!;
    public NutritionFacts NutritionFacts { get; set; } = new();

    INutritionFacts IProduct.NutritionFacts
    {
        get => NutritionFacts;
        set => NutritionFacts = value as NutritionFacts ?? new NutritionFacts
        {
            Carbohydrates = value.Carbohydrates,
            Fat = value.Fat,
            Proteins = value.Proteins,
            Calories = value.Calories
        };
    }
}